// typesafe-di/lib/internal/design.spec.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var design_1 = require("./design");
describe('Design', function () {
    var resolveKey1 = function () { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        return [2 /*return*/, 123];
    }); }); };
    var resolveKey2 = function (injector) { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        return [2 /*return*/, injector.key1.then(function (n) { return "key1 is " + n; })];
    }); }); };
    var resolveKey3 = function (injector) { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
        return [2 /*return*/, injector.key2.then(function (s) { return s === 'key1 is 123'; })];
    }); }); };
    describe('bind', function () {
        it('infers existing container type as injectoable by default', function () { return __awaiter(void 0, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, design_1.Design.bind('key1', function () { return 123; })
                            .bind('key2', function () { return 'test'; })
                            .bind('inferred', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
                            switch (_b.label) {
                                case 0: return [4 /*yield*/, injector.key1];
                                case 1:
                                    _a = (_b.sent()) + "-";
                                    return [4 /*yield*/, injector.key2];
                                case 2: return [2 /*return*/, _a + (_b.sent())];
                            }
                        }); }); })
                            .resolve({})];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        }); });
        it('takes explicit injector type parameter', function () { return __awaiter(void 0, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, design_1.Design.bind('key1', function () { return 123; })
                            .bind('key2', function () { return 'test'; })
                            .bind('explicit', function (injector) { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
                            switch (_a.label) {
                                case 0: return [4 /*yield*/, injector.key0];
                                case 1: return [2 /*return*/, _a.sent()];
                            }
                        }); }); })
                            .resolve({ key0: 0 })];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        }); });
    });
    describe('resolve', function () {
        describe('with valid dependencies', function () {
            it('resolves design', function () { return __awaiter(void 0, void 0, void 0, function () {
                var design, container;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            design = design_1.Design.bind('key1', resolveKey1).bind('key2', resolveKey2);
                            return [4 /*yield*/, design.resolve({})];
                        case 1:
                            container = (_a.sent()).container;
                            expect(container).toEqual({ key1: 123, key2: 'key1 is 123' });
                            return [2 /*return*/];
                    }
                });
            }); });
            describe('with merge', function () {
                it('resolves with merged design', function () { return __awaiter(void 0, void 0, void 0, function () {
                    var design1, design2, container;
                    return __generator(this, function (_a) {
                        switch (_a.label) {
                            case 0:
                                design1 = design_1.Design.bind('key1', resolveKey1).bind('key2', resolveKey2);
                                design2 = design_1.Design.bind('key3', resolveKey3);
                                return [4 /*yield*/, design1.merge(design2).resolve({})];
                            case 1:
                                container = (_a.sent()).container;
                                expect(container).toEqual({
                                    key1: 123,
                                    key2: 'key1 is 123',
                                    key3: true,
                                });
                                return [2 /*return*/];
                        }
                    });
                }); });
            });
        });
        it('resolves bound items exactly once', function () { return __awaiter(void 0, void 0, void 0, function () {
            var counter, design, container;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        counter = 0;
                        design = design_1.Design.bind('sideEffect', function () {
                            counter += 1;
                            return 10;
                        })
                            .bind('dependent1', function (injector) {
                            return injector.sideEffect.then(function (num) { return num + " from dependent1"; });
                        })
                            .bind('dependent2', function (injector) {
                            return injector.sideEffect.then(function (num) { return num + " from dependent2"; });
                        });
                        return [4 /*yield*/, design.resolve({})];
                    case 1:
                        container = (_a.sent()).container;
                        expect(counter).toEqual(1);
                        expect(container).toEqual({
                            sideEffect: 10,
                            dependent1: '10 from dependent1',
                            dependent2: '10 from dependent2',
                        });
                        return [2 /*return*/];
                }
            });
        }); });
        it('throws an error for self dependencies', function () { return __awaiter(void 0, void 0, void 0, function () {
            var design;
            return __generator(this, function (_a) {
                design = design_1.Design.bind('self', function (r) { return r.self; });
                return [2 /*return*/, design.resolve({}).then(function () {
                        throw new Error('should fail');
                    }, function (e) {
                        expect(e.message).toEqual('failed to resolve "self" because: cyclic dependency detected: self -> self');
                    })];
            });
        }); });
        it('throws an error for cyclic dependencies', function () { return __awaiter(void 0, void 0, void 0, function () {
            var design;
            return __generator(this, function (_a) {
                design = design_1.Design.bind('key1', function (r) { return r.key2; }).bind('key2', function (r) { return r.key1; });
                return [2 /*return*/, design.resolve({}).then(function () {
                        throw new Error('should fail');
                    }, function (e) {
                        expect(e.message).toEqual('failed to resolve "key2" because: cyclic dependency detected: key2 -> key1 -> key2');
                    })];
            });
        }); });
        it('can finalize according to the dependent order', function () { return __awaiter(void 0, void 0, void 0, function () {
            var called, callWith, design, finalize;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        called = [];
                        callWith = function (key, value) { return function (item) { return __awaiter(void 0, void 0, void 0, function () {
                            return __generator(this, function (_a) {
                                expect(item).toEqual(value);
                                called.push(key);
                                return [2 /*return*/];
                            });
                        }); }; };
                        design = design_1.Design.bind('key1', resolveKey1, callWith('key1', 123))
                            .bind('key2', resolveKey2, callWith('key2', 'key1 is 123'))
                            .bind('key3', resolveKey3, callWith('key3', true));
                        return [4 /*yield*/, design.resolve({})];
                    case 1:
                        finalize = (_a.sent()).finalize;
                        return [4 /*yield*/, finalize()];
                    case 2:
                        _a.sent();
                        expect(called).toEqual(['key3', 'key2', 'key1']);
                        return [2 /*return*/];
                }
            });
        }); });
        it('cannot call finalize twice', function () { return __awaiter(void 0, void 0, void 0, function () {
            var finalize;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, design_1.Design.empty.resolve({})];
                    case 1:
                        finalize = (_a.sent()).finalize;
                        return [4 /*yield*/, finalize()];
                    case 2:
                        _a.sent();
                        finalize().then(function () {
                            throw new Error('should fail');
                        }, function (e) { return expect(e.message).toEqual('already finalized'); });
                        return [2 /*return*/];
                }
            });
        }); });
        it('can use own error handler', function () { return __awaiter(void 0, void 0, void 0, function () {
            var finalize;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, design_1.Design.bind('key1', function () { return 123; }, function () { return __awaiter(void 0, void 0, void 0, function () {
                            return __generator(this, function (_a) {
                                throw new Error('fails');
                            });
                        }); }).resolve({})];
                    case 1:
                        finalize = (_a.sent()).finalize;
                        return [4 /*yield*/, finalize(function (ps) { return __awaiter(void 0, void 0, void 0, function () {
                                var e_1;
                                return __generator(this, function (_a) {
                                    switch (_a.label) {
                                        case 0:
                                            _a.trys.push([0, 2, , 3]);
                                            return [4 /*yield*/, Promise.all(ps)];
                                        case 1:
                                            _a.sent();
                                            return [3 /*break*/, 3];
                                        case 2:
                                            e_1 = _a.sent();
                                            expect(e_1.message).toEqual('fails');
                                            return [3 /*break*/, 3];
                                        case 3: return [2 /*return*/];
                                    }
                                });
                            }); })];
                    case 2:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        }); });
    });
    describe('with long dependencies', function () {
        var design = design_1.Design.bind('dep1', function () { return __awaiter(void 0, void 0, void 0, function () { return __generator(this, function (_a) {
            return [2 /*return*/, 1];
        }); }); })
            .bind('dep2', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep1];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep3', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep2];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep4', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep3];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep5', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep4];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep6', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep5];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep7', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep6];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep8', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep7];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep9', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep8];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep10', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep9];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep11', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep10];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep12', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep11];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep13', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep12];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep14', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep13];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep15', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep14];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep16', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep15];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep17', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep16];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep18', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep17];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep19', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep18];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep20', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep19];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep21', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep20];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep22', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep21];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep23', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep22];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep24', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep23];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep25', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep24];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep26', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep25];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep27', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep26];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep28', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep27];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep29', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep28];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep30', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep29];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep31', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep30];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep32', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep31];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep33', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep32];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep34', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep33];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep35', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep34];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep36', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep35];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep37', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep36];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep38', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep37];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep39', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep38];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep40', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep39];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep41', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep40];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep42', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep41];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep43', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep42];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep44', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep43];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep45', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep44];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep46', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep45];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep47', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep46];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep48', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep47];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep49', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep48];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep50', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep49];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep51', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep50];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep52', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep51];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep53', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep52];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep54', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep53];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep55', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep54];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep56', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep55];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep57', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep56];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep58', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep57];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep59', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep58];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep60', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep59];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep61', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep60];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep62', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep61];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep63', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep62];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep64', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep63];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep65', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep64];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep66', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep65];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep67', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep66];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep68', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep67];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep69', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep68];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep70', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep69];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep71', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep70];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep72', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep71];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep73', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep72];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep74', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep73];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep75', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep74];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep76', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep75];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep77', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep76];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep78', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep77];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep79', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep78];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep80', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep79];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep81', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep80];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep82', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep81];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep83', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep82];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep84', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep83];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep85', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep84];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep86', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep85];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep87', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep86];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep88', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep87];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep89', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep88];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep90', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep89];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep91', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep90];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep92', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep91];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep93', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep92];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep94', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep93];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep95', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep94];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep96', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep95];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep97', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep96];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep98', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep97];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep99', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep98];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep100', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep99];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep101', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep100];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep102', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep101];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep103', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep102];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep104', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep103];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep105', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep104];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep106', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep105];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep107', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep106];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep108', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep107];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep109', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep108];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep110', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep109];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep111', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep110];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep112', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep111];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep113', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep112];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep114', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep113];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep115', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep114];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep116', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep115];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep117', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep116];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep118', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep117];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep119', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep118];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep120', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep119];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep121', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep120];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep122', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep121];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep123', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep122];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep124', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep123];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep125', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep124];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep126', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep125];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep127', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep126];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep128', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep127];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep129', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep128];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep130', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep129];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep131', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep130];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep132', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep131];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep133', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep132];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep134', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep133];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep135', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep134];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep136', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep135];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep137', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep136];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep138', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep137];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep139', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep138];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep140', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep139];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep141', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep140];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep142', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep141];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep143', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep142];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep144', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep143];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep145', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep144];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep146', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep145];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep147', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep146];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep148', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep147];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep149', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep148];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep150', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep149];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep151', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep150];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep152', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep151];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep153', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep152];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep154', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep153];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep155', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep154];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep156', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep155];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep157', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep156];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep158', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep157];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep159', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep158];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep160', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep159];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep161', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep160];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep162', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep161];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep163', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep162];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep164', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep163];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep165', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep164];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep166', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep165];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep167', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep166];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep168', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep167];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep169', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep168];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep170', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep169];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep171', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep170];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep172', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep171];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep173', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep172];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep174', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep173];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep175', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep174];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep176', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep175];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep177', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep176];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep178', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep177];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep179', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep178];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep180', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep179];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep181', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep180];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep182', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep181];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep183', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep182];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep184', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep183];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep185', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep184];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep186', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep185];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep187', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep186];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep188', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep187];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep189', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep188];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep190', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep189];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep191', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep190];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep192', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep191];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep193', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep192];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep194', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep193];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep195', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep194];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep196', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep195];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep197', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep196];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep198', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep197];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep199', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep198];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('dep200', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    _a = 1;
                    return [4 /*yield*/, injector.dep199];
                case 1: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); })
            .bind('result', function (injector) { return __awaiter(void 0, void 0, void 0, function () { var _a; return __generator(this, function (_b) {
            switch (_b.label) {
                case 0: return [4 /*yield*/, injector.base];
                case 1:
                    _a = (_b.sent());
                    return [4 /*yield*/, injector.dep200];
                case 2: return [2 /*return*/, _a + (_b.sent())];
            }
        }); }); });
        it('works', function () { return __awaiter(void 0, void 0, void 0, function () {
            var container;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, design.resolve({ base: 1000 })];
                    case 1:
                        container = (_a.sent()).container;
                        expect(container.result).toBe(1200);
                        return [2 /*return*/];
                }
            });
        }); });
    });
});
